mds/provider.go

```diff
@@ -135,20 +205,60 @@ func (p *mdsProvider) Configure(ctx context.Context, req provider.ConfigureReque
 		)
 	}
 
+	if clientId == "" && config.Type.ValueString() == oauth_type.ClientCredentials {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("client_id"),
+			"Missing MDS API Client Id",
+			"The provider cannot create the MDS API client as there is a missing or empty value for the MDS API Client Id. "+
+				"Set the password value in the configuration or use the MDS_CLIENT_ID environment variable. "+
+				"If either is already set, ensure the value is not empty.",
+		)
+	}
+
+	if clientSecret == "" && config.Type.ValueString() == oauth_type.ClientCredentials {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("client_secret"),
+			"Missing MDS API Client Secret",
+			"The provider cannot create the MDS API client as there is a missing or empty value for the MDS API Client Secret. "+
+				"Set the password value in the configuration or use the MDS_CLIENT_SECRET environment variable. "+
+				"If either is already set, ensure the value is not empty.",
+		)
+	}
+
+	if orgId == "" && config.Type.ValueString() == oauth_type.ClientCredentials {
```

Comment:
put a parent if for `config.Type.ValueString() == oauth_type.ClientCredentials` then individual checks